.github/workflows/benchmarking.yml

```diff
@@ -94,10 +94,69 @@ jobs:
           name: DerivedData-Xcode
       - run: npm install -g saucectl@0.186.0
       - name: Run Benchmarks in SauceLab
+        id: run-benchmarks-in-sauce-lab
+        continue-on-error: true
         env:
           SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
           SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
-        run: saucectl run --select-suite "${{matrix.suite}}" --config .sauce/benchmarking-config.yml --tags benchmark
+        run: |
+          set -o pipefail && saucectl run \
+            --select-suite "${{matrix.suite}}" \
+            --config .sauce/benchmarking-config.yml \
+            --tags benchmark \
+            --verbose \
+            2>&1 | tee output.log
+
+      - name: Recovery - Extract Test ID from output
+        id: should-retry-test
+        if: ${{ steps.run-benchmarks-in-sauce-lab.outcome == 'failure' }}
+        uses: actions/github-script@v7
+        env:
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+        with:
+          script: |
+            const fs = require('fs');
+            const { execSync } = require('child_process');
+            
+            console.log("Extracting test ID from output log");
+            const outputLog = fs.readFileSync('output.log', 'utf8');
+            const match = outputLog.match(/https:\/\/app\.saucelabs\.com\/tests\/([^\s]+)/);
+            const testId = match?.[1] ?? '';
+            
+            if (!testId) {
+              console.log("No test ID found in the output, retrying...");
+              core.setOutput('RETRY_TEST', 'true');
```

Comment:
`m`: What happens if the log output of saucelabs changes and we can't find a match for every test run anymore? Then we would retry for every run, if I'm not mistaken. If this is true, do you have any ideas on how we can mitigate this problem? Otherwise, it could be that we retry for a long time until we realize that this changed.

Reply:
This is a fair feedback point I do not have a solution for right now.
We could add a GitHub job annotation warning that the retry mechanism was triggered, so at least we have a prominent alert.

Reply:
Then I think adding this downside as a comment and modifying the log message to hint that the log output could have changed if it always retries could mitigate this a bit.

Reply:
I added a warning log statement:

https://github.com/getsentry/sentry-cocoa/actions/runs/12785908532/job/35642172190#step:6:49

It also adds an annotation to the log:

https://github.com/getsentry/sentry-cocoa/actions/runs/12785908532

Does this match your request?

Reply:
As discussed we will leave it like this for now